src/clients/api_client.rs

```diff
@@ -261,33 +271,48 @@ impl ApiClient {
     }
 }
 
-pub fn claim(property: &str, datavalue: json::JsonValue) -> Option<json::JsonValue> {
-    Some(object! {
-        "mainsnak" => object!{
-            "snaktype" => "value",
-            "property" => property,
-            "datavalue" => datavalue
+pub fn claim(property: &str, datavalue: serde_json::Value) -> Option<serde_json::Value> {
+    Some(json!({
+        "mainsnak": {
+            "snaktype": "value",
+            "property": property,
+            "datavalue": datavalue
         },
-        "type" => "statement",
-        "rank" => "normal"
-    })
+        "type": "statement",
+        "rank": "normal"
+    }))
 }
-pub fn claim_string(property: &str, value: &str) -> Option<json::JsonValue> {
+pub fn claim_string(property: &str, value: &str) -> Option<serde_json::Value> {
     let value = value.trim();
     if value.is_empty() {
         // it's impossible to add a claim with an empty value, so we skip it
         None
     } else {
-        claim(property, object! { "value" => value, "type" => "string" })
+        claim(property, json!({ "value": value, "type": "string" }))
     }
 }
 
-pub fn claim_item(property: &str, id: &str) -> Option<json::JsonValue> {
+pub fn claim_item(property: &str, id: &str) -> Option<serde_json::Value> {
     claim(
         property,
-        object! {
-            "value" => object!{ "entity-type" => "item", "id" => id },
-            "type" => "wikibase-entityid",
-        },
+        json!({
+            "value":{ "entity-type": "item", "id": id },
+            "type": "wikibase-entityid",
+        }),
+    )
+}
+
+pub fn claim_coord(property: &str, lon: f64, lat: f64) -> Option<serde_json::Value> {
+    claim(
+        property,
+        json!({
+            "value": {
+                "latitude": lat,
+                "longitude": lon,
+                "precision": 0.001,
```

Comment:
What does the precision mean? 0,001° is about 100m. The stops are rarely precise, but not *that* much off ;)

Reply:
yes, I put a random value without thinking 🙄